pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/flags.go

```diff
@@ -113,125 +101,56 @@ func ConfigToFlags(kubeAPIServerConfig *kubecontrolplanev1.KubeAPIServerConfig)
-	setIfUnset(args, "kubelet-client-certificate", kubeAPIServerConfig.KubeletClientInfo.CertFile)
-	setIfUnset(args, "kubelet-client-key", kubeAPIServerConfig.KubeletClientInfo.KeyFile)
-	setIfUnset(args, "kubelet-https", "true")
-	setIfUnset(args, "kubelet-preferred-address-types", "Hostname", "InternalIP", "ExternalIP")
-	setIfUnset(args, "kubelet-read-only-port", "0")
-	setIfUnset(args, "kubernetes-service-node-port", "0")
-	setIfUnset(args, "max-mutating-requests-inflight", fmt.Sprintf("%d", kubeAPIServerConfig.ServingInfo.MaxRequestsInFlight/2))
-	setIfUnset(args, "max-requests-inflight", fmt.Sprintf("%d", kubeAPIServerConfig.ServingInfo.MaxRequestsInFlight))
-	setIfUnset(args, "min-request-timeout", fmt.Sprintf("%d", kubeAPIServerConfig.ServingInfo.RequestTimeoutSeconds))
-	setIfUnset(args, "proxy-client-cert-file", kubeAPIServerConfig.AggregatorConfig.ProxyClientInfo.CertFile)
-	setIfUnset(args, "proxy-client-key-file", kubeAPIServerConfig.AggregatorConfig.ProxyClientInfo.KeyFile)
-	setIfUnset(args, "requestheader-allowed-names", kubeAPIServerConfig.AuthConfig.RequestHeader.ClientCommonNames...)
-	setIfUnset(args, "requestheader-client-ca-file", kubeAPIServerConfig.AuthConfig.RequestHeader.ClientCA)
-	setIfUnset(args, "requestheader-extra-headers-prefix", kubeAPIServerConfig.AuthConfig.RequestHeader.ExtraHeaderPrefixes...)
-	setIfUnset(args, "requestheader-group-headers", kubeAPIServerConfig.AuthConfig.RequestHeader.GroupHeaders...)
-	setIfUnset(args, "requestheader-username-headers", kubeAPIServerConfig.AuthConfig.RequestHeader.UsernameHeaders...)
-	setIfUnset(args, "secure-port", portString)
-	setIfUnset(args, "service-cluster-ip-range", kubeAPIServerConfig.ServicesSubnet)
-	setIfUnset(args, "service-node-port-range", kubeAPIServerConfig.ServicesNodePortRange)
-	setIfUnset(args, "storage-backend", "etcd3")
-	setIfUnset(args, "storage-media-type", "application/vnd.kubernetes.protobuf")
-	setIfUnset(args, "tls-cert-file", kubeAPIServerConfig.ServingInfo.CertFile)
-	setIfUnset(args, "tls-cipher-suites", kubeAPIServerConfig.ServingInfo.CipherSuites...)
-	setIfUnset(args, "tls-min-version", kubeAPIServerConfig.ServingInfo.MinTLSVersion)
-	setIfUnset(args, "tls-private-key-file", kubeAPIServerConfig.ServingInfo.KeyFile)
-	setIfUnset(args, "tls-sni-cert-key", sniCertKeys(kubeAPIServerConfig.ServingInfo.NamedCertificates)...)
-	setIfUnset(args, "secure-port", portString)
-
-	var keys []string
-	for key := range args {
-		keys = append(keys, key)
-	}
-	sort.Strings(keys)
-
-	var arguments []string
-	for _, key := range keys {
-		for _, token := range args[key] {
-			arguments = append(arguments, fmt.Sprintf("--%s=%v", key, token))
-		}
-	}
-	return arguments, nil
-}
-
-func auditFlags(kubeAPIServerConfig *kubecontrolplanev1.KubeAPIServerConfig) map[string][]string {
-	args := map[string][]string{}
-
-	for key, slice := range kubeAPIServerConfig.APIServerArguments {
-		if !strings.HasPrefix("audit-", key) {
```

Comment:
@soltysh the bug was here already. Just copied (and fixed now).

Reply:
Right, but beforehand that method was only for audit, that's why it has hardcoded value of `audit-`.